rest-server/engines/src/main/scala/com/actionml/engines/ur/URAlgorithm.scala

```diff
@@ -270,7 +262,7 @@ class URAlgorithm private (engine: UREngine, initParams: String, dataset: URData
     }
     */
 
-    SparkContextSupport.getSparkContext(initParams, defaults).map { sc => // or implicit sc =>
+    SparkContextSupport.getSparkContext(initParams, engineId).map { sc => // or implicit sc =>
```

Comment:
Minor point -- it might be good to use the val name `appName = engineId` to self-document a bit.